### PR TITLE
More focus configuration options

### PR DESCRIPTION
**Pull request by mautamu**
Howdy,

As mentioned in #317, it would be nice to tell LeftWM not to update focus on hover (so-called "Sloppy Focus"). The implementations in the literature generally follow one of these patterns:

1. *Sloppy Focus.* Focus follows the mouse, hovering over a window brings it to focus. 
2. *Click-to-Focus.* Focus follows the mouse, but only clicks change focus.
3. *Driven Focus.* Focus disregards the mouse, only keyboard actions drive the focus.
4. *Event Focus.* Focuses when requested by the window/new windows. 

As discussed in #317, I have implemented (3) and (4) by adding two configuration options to the configuration:
```toml
# Defaults
focus_tracks_mouse = true
focus_new_windows = true
```
Both are pretty straightforward. The `focus_tracks_mouse=false` condition unfortunately leaves floating windows unfocusable. It would be nice if we could work to implement (2) and/or rewrite this as a state machine. I will open that as its own issue.

Best,

Mautamu
-----
CC: @retamaut

### CODE REVIEW

File: src/handlers/command_handler.rs [hunk 1]
```diff
@@ -316,28 +316,44 @@ fn move_window_top(
...
         let window_group = &to_reorder[..=index];
         handle = helpers::relative_find(&window_group, is_handle, -val)?.handle;
     } else if let Some(new_focused) = helpers::relative_find(&to_reorder, is_handle, val) {
-        handle = new_focused.handle;
+        focus_handler::focus_window(&mut manager, &handle);
+        let act = DisplayAction::MoveMouseOver(new_focused.handle);
+        manager.actions.push_back(act);
+        let new_window_handle = new_focused.handle;
+        manager.windows.append(&mut to_reorder);
+        return Some(focus_handler::focus_window(manager, &new_window_handle));
```
Comment (AethanFoot) [inline src/handlers/command_handler.rs:356]: Why are we focusing the current window the immediately focusing the new window?

File: src/handlers/command_handler.rs [hunk 2]
```diff
@@ -316,28 +316,44 @@ fn move_window_top(
...
-        // Not the best solution but results
-        // in desired behaviour
-        handle = helpers::relative_find(&to_reorder, is_handle, -val)?.handle;
-        let _ = helpers::cycle_vec(&mut to_reorder, val);
-    } else if let Some(Layout::MainAndDeck) = layout {
-        if to_reorder.len() == 1_usize {
-            return None;
+    // For Monocle we want to also move windows up/down
+    // Not the best solution but results
+    // in desired behaviour
```
[the remark this answers is not in the document]
Comment (AethanFoot) [reply src/handlers/command_handler.rs:328]: This was suppose to highlight the focus_window_change bit

[the remark this answers is not in the document]
Comment (mautamu) [reply src/handlers/command_handler.rs]: I'm not really sure how exactly that happened. I was working off of an older copy and tried to rebase, must've mixed something up. I have fixed this, which also fixed the following comment I made earlier:
> The focus_tracks_mouse=false condition unfortunately leaves floating windows unfocusable.

File: src/handlers/command_handler.rs (hunk 1 above)
Comment (mautamu) [reply src/handlers/command_handler.rs]: The way the code *had* been written, if you didn't do that then focusing by keyboard only wouldn't work at all. #305 preemptively fixed that issue too :laughing:, if only I hadn't undone it.

**Pull request merged**